ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -1103,6 +1093,16 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
           }
         }
       }
+      // 3. check if partition exists on this node and that the store for this partition is available
+      if (storageManager.getStore(partition) == null) {
+        if (partitionsInCurrentNode.contains(partition)) {
+          metrics.replicaUnavailableError.inc();
+          return ServerErrorCode.Replica_Unavailable;
```

Comment:
@jsjtzyy When a replica is down, we would like other nodes (both router and peer servers) to mark it as down (which it will do now with the response handler and in the future with the hybrid scheme of reponse handler + cluster manager notifications). However, I am not sure if replication removes down replicas when it makes a batch replication request to its peers. If it does not, then a request is made continuously for the down replica as well and the validation for the down replica will fail every time here. Could you check if that is the case?

Reply:
I have checked the code and replied in email which might partially solved your problem. Let's talk offline if necessary.